### PR TITLE
fix(*): return non-zero exit code on test failure

### LINKED ISSUE #2089: Release test was timeout but exit code return 0
I think that exit code had better return not 0 when timeout occurs.

```
(%-U-)< helm test hogehoge-release --timeout 10
RUNNING: hogehoge-release
UNKNOWN: hogehoge-release: timed out waiting for the condition

(%-U-)< echo $?
0
```

### PR DESCRIPTION
**Pull request by michelleN**
* resolves #2089, #2166

### CODE REVIEW

**Comment by michelleN** [comment]
I would appreciate some manual testing on this PR. 
Also, I chose to exit cleanly when there are no tests found in the release. Is that a decision we agree with or do we feel like helm test should fail (exit non-zero) when there are no tests found in the release object?

File: cmd/helm/helm_test.go [hunk 1]
```diff
@@ -198,7 +200,27 @@ func (c *fakeReleaseClient) ReleaseHistory(rlsName string, opts ...helm.HistoryO
 }
 
 func (c *fakeReleaseClient) RunReleaseTest(rlsName string, opts ...helm.ReleaseTestOption) (<-chan *rls.TestReleaseResponse, <-chan error) {
-	return nil, nil
+
+	results := make(chan *rls.TestReleaseResponse, len(c.responses))
+	errc := make(chan error, 1)
+
+	go func() {
+		var wg sync.WaitGroup
```
Comment (fibonacci1729) [inline cmd/helm/helm_test.go]: Optionally, you could just add ```wg.Add(len(c.responses))```.

**Review by fibonacci1729 — APPROVED** [review]

File: cmd/helm/release_testing.go [hunk 2]
```diff
@@ -75,16 +76,36 @@ func (t *releaseTestCmd) run() (err error) {
...
 			fmt.Fprintf(t.out, res.Msg+"\n")
+
 		}
 	}
+
+}
+
+type testErr struct {
+	failed int
+	// message []string{}
```
Comment (adamreese) [inline cmd/helm/release_testing.go]: delete me

**Review by adamreese — CHANGES_REQUESTED** [review]
Run a `make bootstrap protoc` to ensure correct versions for code generation

```
helm on pr/2425
❯ make protoc
...

helm on pr/2425 [!]
❯ git status
On branch pr/2425
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   pkg/proto/hapi/chart/chart.pb.go
	modified:   pkg/proto/hapi/chart/config.pb.go
	modified:   pkg/proto/hapi/chart/metadata.pb.go
	modified:   pkg/proto/hapi/chart/template.pb.go
	modified:   pkg/proto/hapi/release/hook.pb.go
	modified:   pkg/proto/hapi/release/info.pb.go
	modified:   pkg/proto/hapi/release/release.pb.go
	modified:   pkg/proto/hapi/release/status.pb.go
	modified:   pkg/proto/hapi/release/test_run.pb.go
	modified:   pkg/proto/hapi/release/test_suite.pb.go
	modified:   pkg/proto/hapi/rudder/rudder.pb.go
	modified:   pkg/proto/hapi/services/tiller.pb.go
	modified:   pkg/proto/hapi/version/version.pb.go

no changes added to commit (use "git add" and/or "git commit -a")
```

**Review by adamreese — APPROVED** [review]

**Pull request merged**